### ISSUE TITLE
Middleware default entrypoints and "watch: true" 

### ISSUE THREAD

**Issue by lfache** [containous/traefik#6441]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I've upgraded to Traefik v2.2-rc1 and set up a simple HTTP -> HTTPS with default middlewares : works fine. But if you use a middleware with `@file` , "hot" reload with `watch: true` break the https for the service.

### What did you expect to see?

Hot reload create only http entrypoint for the service.

I have try "traefik.http.routers.ghost.entrypoints=http, https" or "traefik.http.routers.ghost.entrypoints=https, http"

But same behavior.

### What did you see instead?

Create entrypoints for http and https.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.2.0-rc1
Codename:     cantal
Go version:   go1.14
Built:        2020-03-05T19:10:06Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

traefik.yml : 

```yml
entryPoints:
  http:
    address: :80
    http:
      redirections:
        entryPoint:
          to: https
          scheme: https

  https:
    address: :443
    http:
      tls:
        certResolver: myresolver
      middlewares:
      - SecHeader@file
 
providers:
  docker:
    exposedByDefault: false

  file:
    directory: /etc/traefik/custom/
    watch: true

certificatesResolvers:
  myresolver:
    acme:
      email: "XXXX"
      storage: "/letsencrypt/acme.json"
      httpChallenge:
        entryPoint: http

```

custom/middlewares.yml : 

```yml
http:
  middlewares:
    SecHeader:
      headers:
        frameDeny: true
        contentTypeNosniff: true
        browserXssFilter: true

```
docker-compose.yml : 

```yml
version: '3.7'
services:
  traefik:
    image: traefik:2.2
    container_name: traefik
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./traefik.yml:/etc/traefik/traefik.yml:ro
      - ./custom/:/etc/traefik/custom/:ro
      - traefik_ssl:/letsencrypt
    networks:
      - traefik


  ghost:
    depends_on:
      - mysql
      - traefik
    container_name: ghost
    image: ghost:3
    env_file: .ghost.env
    restart: always
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik"
      - "traefik.http.routers.ghost.entrypoints=http, https"
      - "traefik.http.routers.ghost.rule=Host(`XXXX`)"
      - "traefik.http.services.ghost.loadbalancer.server.port=2368"
    volumes:
      - www-data:/var/lib/ghost/content
    networks:
      - traefik
      - lan

```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
time="2020-03-06T14:48:02Z" level=debug msg="Configuration received from provider file: {\"http\":{\"middlewares\":{\"SecHeader\":{\"headers\":{\"customResponseHeaders\":{\"X-Powered-By\":\"\"},\"stsSeconds\":31536000,\"stsIncludeSubdomains\":true,\"forceSTSHeader\":true,\"frameDeny\":true,\"contentTypeNosniff\":true,\"browserXssFilter\":true,\"contentSecurityPolicy\":\"script-src 'self'; img-src 'self'\",\"referrerPolicy\":\"same-origin\",\"featurePolicy\":\"vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'\"}}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=file
time="2020-03-06T14:48:02Z" level=debug msg="Configuration received from provider file: {\"http\":{\"middlewares\":{\"SecHeader\":{\"headers\":{\"customResponseHeaders\":{\"X-Powered-By\":\"\"},\"stsSeconds\":31536000,\"stsIncludeSubdomains\":true,\"forceSTSHeader\":true,\"frameDeny\":true,\"contentTypeNosniff\":true,\"browserXssFilter\":true,\"contentSecurityPolicy\":\"script-src 'self'; img-src 'self'\",\"referrerPolicy\":\"same-origin\",\"featurePolicy\":\"vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'\"}}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=file
time="2020-03-06T14:48:02Z" level=info msg="Skipping same configuration" providerName=file
time="2020-03-06T14:48:02Z" level=debug msg="Adding certificate for domain(s) XXXXXXXX"
time="2020-03-06T14:48:02Z" level=debug msg="No default certificate, generating one"
time="2020-03-06T14:48:02Z" level=debug msg="Configuration received from provider file: {\"http\":{\"middlewares\":{\"SecHeader\":{\"headers\":{\"customResponseHeaders\":{\"X-Powered-By\":\"\"},\"stsSeconds\":31536000,\"stsIncludeSubdomains\":true,\"forceSTSHeader\":true,\"frameDeny\":true,\"contentTypeNosniff\":true,\"browserXssFilter\":true,\"contentSecurityPolicy\":\"script-src 'self'; img-src 'self'\",\"referrerPolicy\":\"same-origin\",\"featurePolicy\":\"vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'\"}}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=file
time="2020-03-06T14:48:02Z" level=debug msg="Configuration received from provider file: {\"http\":{\"middlewares\":{\"SecHeader\":{\"headers\":{\"customResponseHeaders\":{\"X-Powered-By\":\"\"},\"stsSeconds\":31536000,\"stsIncludeSubdomains\":true,\"forceSTSHeader\":true,\"frameDeny\":true,\"contentTypeNosniff\":true,\"browserXssFilter\":true,\"contentSecurityPolicy\":\"script-src 'self'; img-src 'self'\",\"referrerPolicy\":\"same-origin\",\"featurePolicy\":\"vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'\"}}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=file
time="2020-03-06T14:48:02Z" level=info msg="Skipping same configuration" providerName=file
time="2020-03-06T14:48:02Z" level=info msg="Skipping same configuration" providerName=file
time="2020-03-06T14:48:02Z" level=debug msg="Configuration received from provider file: {\"http\":{\"middlewares\":{\"SecHeader\":{\"headers\":{\"customResponseHeaders\":{\"X-Powered-By\":\"\"},\"stsSeconds\":31536000,\"stsIncludeSubdomains\":true,\"forceSTSHeader\":true,\"frameDeny\":true,\"contentTypeNosniff\":true,\"browserXssFilter\":true,\"contentSecurityPolicy\":\"script-src 'self'; img-src 'self'\",\"referrerPolicy\":\"same-origin\",\"featurePolicy\":\"vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'\"}}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=file
time="2020-03-06T14:48:02Z" level=debug msg="Configuration received from provider file: {\"http\":{\"middlewares\":{\"SecHeader\":{\"headers\":{\"customResponseHeaders\":{\"X-Powered-By\":\"\"},\"stsSeconds\":31536000,\"stsIncludeSubdomains\":true,\"forceSTSHeader\":true,\"frameDeny\":true,\"contentTypeNosniff\":true,\"browserXssFilter\":true,\"contentSecurityPolicy\":\"script-src 'self'; img-src 'self'\",\"referrerPolicy\":\"same-origin\",\"featurePolicy\":\"vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'\"}}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=file
time="2020-03-06T14:48:02Z" level=info msg="Skipping same configuration" providerName=file
time="2020-03-06T14:48:02Z" level=info msg="Skipping same configuration" providerName=file
time="2020-03-06T14:48:02Z" level=debug msg="Configuration received from provider file: {\"http\":{\"middlewares\":{\"SecHeader\":{\"headers\":{\"customResponseHeaders\":{\"X-Powered-By\":\"\"},\"stsSeconds\":31536000,\"stsIncludeSubdomains\":true,\"forceSTSHeader\":true,\"frameDeny\":true,\"contentTypeNosniff\":true,\"browserXssFilter\":true,\"contentSecurityPolicy\":\"script-src 'self'; img-src 'self'\",\"referrerPolicy\":\"same-origin\",\"featurePolicy\":\"vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'\"}}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=file
time="2020-03-06T14:48:02Z" level=debug msg="Configuration received from provider file: {\"http\":{\"middlewares\":{\"SecHeader\":{\"headers\":{\"customResponseHeaders\":{\"X-Powered-By\":\"\"},\"stsSeconds\":31536000,\"stsIncludeSubdomains\":true,\"forceSTSHeader\":true,\"frameDeny\":true,\"contentTypeNosniff\":true,\"browserXssFilter\":true,\"contentSecurityPolicy\":\"script-src 'self'; img-src 'self'\",\"referrerPolicy\":\"same-origin\",\"featurePolicy\":\"vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'\"}}}},\"tcp\":{},\"udp\":{},\"tls\":{}}" providerName=file
time="2020-03-06T14:48:02Z" level=info msg="Skipping same configuration" providerName=file
time="2020-03-06T14:48:02Z" level=info msg="Skipping same configuration" providerName=file
time="2020-03-06T14:48:02Z" level=debug msg="Added outgoing tracing middleware noop@internal" routerName=http-to-https@internal middlewareName=tracing middlewareType=TracingForwarder entryPointName=http
time="2020-03-06T14:48:02Z" level=debug msg="Creating middleware" entryPointName=http routerName=http-to-https@internal middlewareName=redirect-http-to-https@internal middlewareType=RedirectScheme
time="2020-03-06T14:48:02Z" level=debug msg="Setting up redirection to https 443" routerName=http-to-https@internal middlewareName=redirect-http-to-https@internal middlewareType=RedirectScheme entryPointName=http
time="2020-03-06T14:48:02Z" level=debug msg="Adding tracing to middleware" routerName=http-to-https@internal middlewareName=redirect-http-to-https@internal entryPointName=http
time="2020-03-06T14:48:02Z" level=debug msg="Creating middleware" middlewareName=pipelining middlewareType=Pipelining routerName=ghost@docker serviceName=ghost entryPointName=http
time="2020-03-06T14:48:02Z" level=debug msg="Creating load-balancer" entryPointName=http routerName=ghost@docker serviceName=ghost
time="2020-03-06T14:48:02Z" level=debug msg="Creating server 0 http://192.168.128.3:2368" serviceName=ghost entryPointName=http serverName=0 routerName=ghost@docker
time="2020-03-06T14:48:02Z" level=debug msg="Added outgoing tracing middleware ghost" middlewareName=tracing middlewareType=TracingForwarder entryPointName=http routerName=ghost@docker
time="2020-03-06T14:48:02Z" level=debug msg="Creating middleware" entryPointName=http middlewareName=traefik-internal-recovery middlewareType=Recovery
time="2020-03-06T14:48:02Z" level=debug msg="Added outgoing tracing middleware api@internal" entryPointName=traefik routerName=api@internal middlewareName=tracing middlewareType=TracingForwarder
time="2020-03-06T14:48:02Z" level=debug msg="Added outgoing tracing middleware dashboard@internal" routerName=dashboard@internal entryPointName=traefik middlewareName=tracing middlewareType=TracingForwarder
time="2020-03-06T14:48:02Z" level=debug msg="Creating middleware" middlewareType=StripPrefix entryPointName=traefik routerName=dashboard@internal middlewareName=dashboard_stripprefix@internal
time="2020-03-06T14:48:02Z" level=debug msg="Adding tracing to middleware" middlewareName=dashboard_stripprefix@internal entryPointName=traefik routerName=dashboard@internal
time="2020-03-06T14:48:02Z" level=debug msg="Creating middleware" routerName=dashboard@internal entryPointName=traefik middlewareName=dashboard_redirect@internal middlewareType=RedirectRegex
time="2020-03-06T14:48:02Z" level=debug msg="Setting up redirection from ^(http:\\/\\/[^:\\/]+(:\\d+)?)\\/$ to ${1}/dashboard/" routerName=dashboard@internal entryPointName=traefik middlewareName=dashboard_redirect@internal middlewareType=RedirectRegex
time="2020-03-06T14:48:02Z" level=debug msg="Adding tracing to middleware" entryPointName=traefik routerName=dashboard@internal middlewareName=dashboard_redirect@internal
time="2020-03-06T14:48:02Z" level=debug msg="Creating middleware" entryPointName=traefik middlewareName=traefik-internal-recovery middlewareType=Recovery
```

**Comment by ldez** [comment]
Thank you very much for testing the RC :heart:

I reproduced and I find the bug.

Thank you :+1:

**Comment by ldez** [comment]
Some notes not related to this bug:
- we recommend to use `web` and `websecure` instead of `http` and `https` because this can be a source of confusion with the protocol.
- we recommend to define only the secured entry point on a router to avoid to expose your service if the redirection failed.

**Comment by traefiker** [comment]
Closed by #6444.
